### PR TITLE
feat: add dataset title when creating a package

### PR DESCRIPTION
**Pull request by ranim-n**

### CODE REVIEW

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/NGSIToCKAN.java [hunk 1]
```diff
@@ -272,7 +272,6 @@ protected void persistFlowFile(final ProcessContext context, final FlowFile flow
         final String dataModel=context.getProperty(DATA_MODEL).getValue();
         final NGSIEvent event=n.getEventFromFlowFile(flowFile,session,ngsiVersion);
         final long creationTime = event.getCreationTime();
-        final String fiwareService = (event.getFiwareService().compareToIgnoreCase("nd")==0)?context.getProperty(DEFAULT_SERVICE).getValue():event.getFiwareService();
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/NGSIToCKAN.java:275]: it was the last remaining use of `DEFAULT_SERVICE`, so you can now safely remove the property

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java [hunk 2]
```diff
@@ -309,9 +310,9 @@ private String createPackage(String pkgName, String orgId, DCATMetadata dcatMeta
         JsonObject dataJson = new JsonObject();
         dataJson.addProperty("name",pkgName);
         dataJson.addProperty("owner_org",orgId);
+        dataJson.addProperty("notes", pkgDescription );
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java]: don't get why you added this `notes` metadata and why it is important for the current PR

Comment (ranim-n) [reply nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java]: the `notes` contains the package description..should I add it in a seperate PR?

Comment (bobeal) [reply nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java]: yes, it is a bit out of the scope of the current PR (and should be dealt with a more general one concerning the metadata of the packages).

in this one, you just have to make sure that the package id and title make use of information from the dataset (and that the title is exactly the title of the dataset, not something with `_` or encoded characters)

Comment (ranim-n) [reply nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java]: for the encoding, uppercase letters are encoded to their hexadecimal unicode value

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/NGSIToCKAN.java [hunk 3]
```diff
@@ -291,7 +280,8 @@ public void aggregate(Entity entity, long creationTime, String dataModel) {
             getLogger().debug("DCAT metadata: {}" , dcatMetadata);
 
             for (Entity entity : entities) {
-                final String pkgName = ckanBackend.buildPkgName(fiwareService,entity,dataModel,enableEncoding,enableLowercase,ngsiVersion,dcatMetadata);
+                final String pkgTitle = ckanBackend.getDataFromRelationshipDetails(entity, "title");
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/NGSIToCKAN.java]: this is not related to CKAN, the `getDataFromRelationshipDetails` method should go elsewhere (in `NGSIUtils`?).

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java [hunk 4]
```diff
@@ -309,10 +310,9 @@ private String createPackage(String pkgName, String orgId, DCATMetadata dcatMeta
         JsonObject dataJson = new JsonObject();
         dataJson.addProperty("name",pkgName);
         dataJson.addProperty("owner_org",orgId);
-
+        dataJson.addProperty("title", NGSIEncoders.encodePostgreSQL(pkgTitle));
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java]: ?!

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/utils/NGSIUtils.java [hunk 5]
```diff
@@ -100,35 +101,57 @@ public void process(final InputStream in) throws IOException {
...
-                                    subAttrName=keyOne;
-                                    subAttrType=value2.get("type").toString();
-                                    if ("Relationship".contentEquals(subAttrType)){
-                                        subAttrValue = value2.get("object").toString();
-                                    }else if ("Property".contentEquals(subAttrType)){
-                                        subAttrValue = value2.get("value").toString();
-                                    }else if ("GeoProperty".contentEquals(subAttrType)){
-                                        subAttrValue = value2.get("value").toString();
+
+                        }else if ("GeoProperty".contentEquals(attrType)){
```
Comment (bobeal) [inline nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/utils/NGSIUtils.java:105]: this is coming from `NgsiLdToPostgreSQL`?

Comment (ranim-n) [reply nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/utils/NGSIUtils.java:105]: yes

File: nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java (hunk 4 above)
Comment (ranim-n) [reply nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/ckan/CKANBackend.java]: The title of the dataset..?

**Review by bobeal — APPROVED** [review]
